#161114862 A user should be able to favourite an article

Victor-Ugwueze opened the pull request:
#### What does this PR do?
- Favourite an article
   
#### Description of Task to be completed?
  - add following methods   to ArticleController   
      - addFavourite   
     - removeFavourite 
     - getAllFavourite

 - add endpoints
    - list all all favourite for user
   ````GET /api/v1/articles/favourites````

   - add article to favourite
   ```` POST /api/v1/articles/:articleId/favourite````

   - remove article from favourite
   ```` DELETE /api/v1/articles/:articleId/favourite````

#### How should this be manually tested?
    - Clone the repo
    - cd into the 'lotus-ah'
    - Run 'npm install'
    - Use the .env.sample format to create a .env file
    - Run 'npm run migrate'
    - Run 'npm run start:dev'

#### Any background context you want to provide?
  It's needed to complete [this feature](https://www.pivotaltracker.com/story/show/160451607 "send notification feature"). As a user should receive updates on his favourited articles.

#### What are the relevant pivotal tracker stories?
[#161114862](https://www.pivotaltracker.com/story/show/160451597 "reset password")
 
### Question
 NA

### Screenshot
 NA

andela-phemmy committed:
ft(auth): configured and initialized social auth

andela-phemmy committed:
ft(api): Enabled social authentication

 - Configured passport integration in config/passport.js file
 - Created social authentication route
 - Created controller for social auth in controllers/SocialAuthController
 - Initialized passport in index.js

feat(Auth): implemented social authentication

 - Created a PassportMockStrategy to mock response
 - Implemented the passport for social authentication

Delivers #161081708

fix: implement TTL feedback

 - align Jsdoc properly
 - add new line to eslintignore
 - removed unused line from index
  - remove blank lines

Victor-Ugwueze committed:
send notification

Victor-Ugwueze committed:
send notification

Victor-Ugwueze committed:
feat(favorite article): user favorite articles

- define favouriteArticle.test.js file

    the test cases covers the following:
      *  adding an article to favourite
      *  removing an article from favourite
      *  get all favourite article for user

- add faouriteArticle model
- add methods this 3 methods to ArticleController
   1. addfavourite
   2. removeFavourite
   3. getAllFavourites

[Delivers #161114862]

Victor-Ugwueze committed:
feat(favorite article): user favorite articles

- define favouriteArticle.test.js file

    the test cases covers the following:
      *  adding an article to favourite
      *  removing an article from favourite
      *  get all favourite article for user

- add faouriteArticle model
- add methods this 3 methods to ArticleController
   1. addfavourite
   2. removeFavourite
   3. getAllFavourites

[Delivers #161114862]

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile
User shoulld not be able to update other user's profile
closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile)
User should be able to update his profile
User should be able to view other user's profile
User shoulld not be able to update other user's profile
closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile)
User should be able to update his profile
User should be able to view other user's profile
User shoulld not be able to update other user's profile
closes 160451599

KaiserPhemi commented on server/controllers/ArticleController.js:
```diff
@@ -272,5 +274,115 @@ class ArticleController {
+        },
+      ],
+      where: { userId, }
+    }).then(result => res.status(200).json({ status: 'Success', result }))
```
#### Can we not make this a one-liner please?
This should read:
```js
.then(result => {
  res.status(200)
      .json({
        status: 'Success',
        result
      })
})
```
This also apply to the next line of code and every where in the code base that this was implemented.

KaiserPhemi replied:
@Victor-Ugwueze 
You did not fix this??!!!

Victor-Ugwueze committed:
fix(style response): fix json object response style

fix json object respose on favourite article methods
  * addFavourite
 * removeFavourite
 * getAllFavorite

Victor-Ugwueze committed:
fix(style response): fix json object response style

fix json object respose on favourite article methods
  * addFavourite
 * removeFavourite
 * getAllFavorite

Victor-Ugwueze commented:
I have implemented the feedback @KaiserPhemi. Thank you

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

obulaworld reviewed (approved):
LGTM

KaiserPhemi commented:
@Victor-Ugwueze 
👎

KaiserPhemi committed:
Merge pull request #21 from andela/ft-enable-social-login

#111504508 Enable social authentication (AUTH)

Victor-Ugwueze commented:
@KaiserPhemi, Please I am sorry. Can you look at the screenshot below

<img width="1440" alt="screen shot 2018-10-11 at 3 12 19 pm" src="https://user-images.githubusercontent.com/24890605/46810287-4011c780-cd68-11e8-9430-48fd66100d3d.png">

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

obulaworld committed:
feat(user profile) User should be able to update Profile
User should be able to update his profile
User should be able to view other user's profile

closes 160451599

KaiserPhemi committed:
Merge pull request #22 from andela/feat-profile-160451599

#160451599 User should be able to update his Profile

KaiserPhemi commented:
@Victor-Ugwueze 
Please fix merge conflicts.

Victor-Ugwueze committed:
fix(style response): fix json object response style

- fix response object style on all methods of ArticleController

Victor-Ugwueze committed:
feat(favorite article): user favorite articles

 - fix merge conflict
